### PR TITLE
Created Linux Gui that writes to scoring.conf

### PR DESCRIPTION
**Pull request by aaditgupta21**
Linux Gui with zenity that writes to scoring.conf

### CODE REVIEW

File: misc/gui_linux.sh [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+echo "name = \"$name\"" >> /opt/aeacus/scoring.conf
+echo "title = \"$title\"" >> /opt/aeacus/scoring.conf
+echo "user = \"$user\"" >> /opt/aeacus/scoring.conf
+echo "os = \"$os\"" >> /opt/aeacus/scoring.conf
+echo "remote = \"$remote\"" >> /opt/aeacus/scoring.conf
+echo "local = \"$local\"" >> /opt/aeacus/scoring.conf
+echo "enddate = \"$end\"" >> /opt/aeacus/scoring.conf
+echo "nodestory = \"$destroy\"" >> /opt/aeacus/scoring.conf
+
+printf "$vulns" >> /opt/aeacus/scoring.conf
```
Comment (safinsingh) [inline misc/gui_linux.sh]: Why is `printf` being used here?

File: misc/gui_linux.sh [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+global=$(zenity --forms --add-entry="Round Name" --add-entry="Round Title" --add-entry="Main User" --add-entry="OS" --add-entry="Remote" --add-entry="Local" --add-entry="EndDate" --add-entry="NoDestroy" --separator=$'\034' --title=AeacusSE)
+
+IFS=$'\034' read -r name title user os remote local end destroy <<< "$global"
+
+next=""
+vulns=""
+
+while [[ "$next" != "None" ]]
+do
```
Comment (safinsingh) [inline misc/gui_linux.sh:10]: Can we use `while [[]]; do` instead here?

File: misc/gui_linux.sh [hunk 3]
```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+
```
Comment (safinsingh) [inline misc/gui_linux.sh]: Extra line

File: misc/gui_linux.sh [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+IFS=$'\034' read -r name title user os remote local end destroy <<< "$global"
+
+next=""
+vulns=""
+
+while [[ "$next" != "None" ]]
+do
+
+next=$(zenity --list --title="Add vulnerabilities" --text "Which category do you want to add?" --radiolist --column "Pick" --column "Vuln Category" FALSE "Command" FALSE "CommandNot" FALSE "FileExists" FALSE "FileExistsNot" FALSE "FileContains" FALSE "FileContainsNot" FALSE "FileContainsRegex" FALSE "FileContainsRegexNot" FALSE "DirContainsRegex" FALSE "DirContainsRegexNot" FALSE "FileEquals" FALSE "FileEqualsNot" FALSE "PackageInstalled" FALSE "PackageInstalledNot" FALSE "ServiceUp" FALSE "ServiceUpNot" FALSE "UserExists" FALSE "UserExistsNot" FALSE "FirewallUp" FALSE "UserInGroup" FALSE "UserInGroupNot" FALSE "GuestDisabledLDM" FALSE "None" --height=700 --width=600 --title=AeacusSE)
```
Comment (safinsingh) [inline misc/gui_linux.sh:12]: Can we get the appropriate tabbing in here?

File: misc/gui_linux.sh [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+echo "user = \"$user\"" >> /opt/aeacus/scoring.conf
+echo "os = \"$os\"" >> /opt/aeacus/scoring.conf
+echo "remote = \"$remote\"" >> /opt/aeacus/scoring.conf
+echo "local = \"$local\"" >> /opt/aeacus/scoring.conf
+echo "enddate = \"$end\"" >> /opt/aeacus/scoring.conf
+echo "nodestory = \"$destroy\"" >> /opt/aeacus/scoring.conf
+
+printf "$vulns" >> /opt/aeacus/scoring.conf
+
+
```
Comment (safinsingh) [inline misc/gui_linux.sh]: Extra line

File: misc/gui_linux.sh [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+	data=$(zenity --forms --text="Pass if User is not in Group" --add-entry "Message" --add-entry="Points" --add-entry="User" --add-entry="Group" --separator=$'\034')
+	IFS=$'\034' read -r mess pts arg1 arg2 <<< "$data"
+	vulns="${vulns}\n[[check]]\nmessage = \"${mess}\"\npoints = \"${pts}\"\n[[check.pass]]\ntype= \"UserInGroupNot\"\narg1 = \"${arg1}\"\narg2 = \"${arg2}\"\n"
+	;;
+"GuestDisabledLDM")
+	data=$(zenity --forms --text="Pass if Gues is Disabled" --separator=$'\034')
+	vulns="${vulns}\n[[check]]\nmessage = \"${mess}\"\npoints = \"${pts}\"\n[[check.pass]]\ntype= \"GuestDisblaedLDM\"\n"
+	;;
+esac
+
```
Comment (safinsingh) [inline misc/gui_linux.sh:136]: Formatting here

**Review by safinsingh — CHANGES_REQUESTED** [review]
See comments

File: misc/gui_linux.sh (hunk 1 above)
Comment (aaditgupta21) [reply misc/gui_linux.sh]: Echo Doesn't do the formatting correctly, for example \n.

File: misc/gui_linux.sh (hunk 4 above)
Comment (aaditgupta21) [reply misc/gui_linux.sh:12]: Can't do that because zenity doesn't support tabs.

File: misc/gui_linux.sh (hunk 6 above)
Comment (aaditgupta21) [reply misc/gui_linux.sh:136]: Zenity Doesn't support Tabs, already tried it

File: misc/gui_linux.sh (hunk 2 above)
Comment (aaditgupta21) [reply misc/gui_linux.sh:10]: Could you elaborate?

File: misc/gui_linux.sh (hunk 1 above)
Comment (safinsingh) [reply misc/gui_linux.sh]: Use `echo -e`

File: misc/gui_linux.sh (hunk 4 above)
Comment (safinsingh) [reply misc/gui_linux.sh:12]: Oh ok got it

**Pull request merged**